overlord/snapstate/snapstate.go

```diff
@@ -1964,7 +1964,11 @@ func autoRefreshPhase1(ctx context.Context, st *state.State) ([]string, []*state
 		if err := checkChangeConflictIgnoringOneChange(st, up.InstanceName(), snapst, fromChange); err != nil {
 			logger.Noticef("cannot refresh snap %q: %v", up.InstanceName(), err)
 		} else {
-			updates = append(updates, up)
+			info, err := snapst.CurrentInfo()
+			if err != nil {
+				return nil, nil, fmt.Errorf("internal error: cannot get current info: %v", err)
+			}
+			updates = append(updates, info)
```

Comment:
it seems that no test has a situation that makes this change visible, it's probably annoying but it would be good to have such a test, easiest maybe is to play with a base change

Reply:
Added, but this actually uncovered that affectedByRefresh already internally uses current info anyway (related to all snapstates lookup) so the test is not good for proving this change (but makes sense to have it anyway). affectedByRefresh should probably just take snap names, I'll experiment with it.

Reply:
I pushed the change to affectedByRefresh signature.